src/java.base/share/classes/sun/nio/ch/ChannelInputStream.java

```diff
@@ -46,7 +46,7 @@
  * @author Mark Reinhold
  */
 class ChannelInputStream extends InputStream {
-    private static final int DEFAULT_BUFFER_SIZE = 8192;
+    static final int DEFAULT_BUFFER_SIZE = 8192;
```

Comment:
Since this buffer size is now being used in more places, is it worth testing with 16KiB to match `InputStream` from https://github.com/openjdk/jdk/pull/11783?

```suggestion
    static final int DEFAULT_BUFFER_SIZE = 16384;
```


https://github.com/openjdk/jdk/blob/11d431b2c436d6b2a0aa7a00d676a93c1b87da0e/src/java.base/share/classes/java/io/InputStream.java#L56

Reply:
I can look into running a benchmark on this.

Reply:
A quick measurement suggests that doubling the buffer size to 16384 would increase `read` throughput by more than 15%, and `write` throughput by more than 40%.

Reply:
That would be a nice additional benefit on top of reducing the direct memory allocations

Reply:
I guess it’s still slower than uncapped. Brian mentioned only 8k vs 16k, not bigger or unconstrained. (And it also depends on the physical drivers below the test system)